ecssd_agent.go

```diff
@@ -371,13 +408,17 @@ func main() {
 	}
 	configuration.HostedZoneId = zoneId
 	metadataClient := ec2metadata.New(session.New())
-	hostname, err := metadataClient.GetMetadata("/hostname")
-	configuration.Hostname = hostname
+	localIP, err := metadataClient.GetMetadata("/local-ipv4")
+	configuration.Hostname = "ip-" + strings.Replace(localIP, ".", "-", 3) + "." + DNSName
```

Comment:
When cleaning up, couldn't you just only look at those records matching the pattern above and skip all others?

If you also want to catch the case that someone has manually created such records, we could use the same mechanism as for the SRV records. ecssd_agent uses Route53's SetIdentifier field to associate records with Docker containers. We do something similar for A records and tag them. Only those records would then be eligible for cleanup.